compiler/extccomp.nim

```diff
@@ -766,14 +766,29 @@ proc callCCompiler*(projectfile: string) =
       add(objfiles, ' ')
       add(objfiles, quoteShell(
           addFileExt(objFile, CC[cCompiler].objExt)))
+
     for x in toCompile:
       let objFile = if noAbsolutePaths(): x.obj.extractFilename else: x.obj
       add(objfiles, ' ')
       add(objfiles, quoteShell(objFile))
 
     linkCmd = getLinkCmd(projectfile, objfiles)
     if optCompileOnly notin gGlobalOptions:
-      execLinkCmd(linkCmd)
+      when defined(windows):
+        let isCmdFileUsed = cCompiler == ccGcc and linkCmd.len >= 32_767
+        let cmdFile = projectfile & "_linker_cmd.txt"
+        if isCmdFileUsed:
+          let gccExeIdx = linkCmd.find("-o")
```

Comment:
This is a hacky post processing step and the non-hacky solution is to change how ``execLinkCmd`` works.